agent/consul/rpc.go

```diff
@@ -419,10 +419,12 @@ func (s *Server) handleConsulConn(conn net.Conn) {
 
 		if err := s.rpcServer.ServeRequest(rpcCodec); err != nil {
 			if err != io.EOF && !strings.Contains(err.Error(), "closed") {
-				s.rpcLogger().Error("RPC error",
-					"conn", logConn(conn),
-					"error", err,
-				)
+				if !errors.Is(err, rate.ErrRetryLater) && !errors.Is(err, rate.ErrRetryElsewhere) {
```

Comment:
It looks like we already have an `if` to filter out noisy error types on the line above, should we collapse it into one?

```Go
skipLog := errors.Is(err, io.EOF) ||
	errors.Is(err, rate.ErrRetryLater) ||
	errors.Is(err, rate.ErrRetryElsewhere) ||
	strings.Contains(err.Error(), "closed")

if !skipLog {
  // ...
}
```

Reply:
@boxofrad, @jmurret 
I changed that to simplify if while keeping the same behaviour. Please give it another 👀